internal/venice-client-common/src/main/java/com/linkedin/venice/exceptions/StorePartitionNotFoundException.java

```diff
@@ -0,0 +1,10 @@
+package com.linkedin.venice.exceptions;
+
+/**
+ * Class for all Venice exceptions that are triggered by resources not being found.
```

Comment:
I see `VeniceNoStoreException` to denote absence of the store and any extensions of it would be representative of extensions of `store`.

e.g., I see `StorepartitionNotFoundException` as an exception being thrown by the server when a query comes for a store and partition that is not hosted by it. However, that doesn't necessarily mean the store is absent, rather the store partition is not hosted by the server. 

If the store was indeed was absent, i would expect `VeniceNoStoreException` at that point. 

Can you please clarify what this exception denotes? The java document points to resources not found but doesn't talk about what those resources are?

Reply:
I missed to update the documents. I could update it to `Class for all Venice exceptions that are triggered by store partitions not being found.`

I think there are some nuances on missing stores, missing versions and missing partitions. 

> If the store was indeed was absent, i would expect VeniceNoStoreException at that point.

This sounds a right idea, so we can throw `VeniceNoStoreException` and depending on whether that's missing, we can throw the exception accordingly